presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java

```diff
@@ -28,20 +30,25 @@
 {
     private final KerberosAuthentication kerberosAuthentication;
 
-    public static KerberosHadoopAuthentication createKerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication, HdfsConfiguration hdfsConfiguration)
+    public static KerberosHadoopAuthentication createKerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication, HdfsConfigurationUpdater updater)
     {
-        KerberosHadoopAuthentication kerberosHadoopAuthentication = new KerberosHadoopAuthentication(kerberosAuthentication);
-
-        Configuration configuration = requireNonNull(hdfsConfiguration, "hdfsConfiguration is null").getConfiguration();
-        initializeConfigs(configuration);
-        return kerberosHadoopAuthentication;
+        initializeConfigs(getConfiguration(updater));
+        return new KerberosHadoopAuthentication(kerberosAuthentication);
     }
 
     private KerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication)
     {
         this.kerberosAuthentication = requireNonNull(kerberosAuthentication, "kerberosAuthentication is null");
     }
 
+    private static Configuration getConfiguration(HdfsConfigurationUpdater updater)
+    {
+        Configuration configuration = new Configuration(false);
+        copy(getInitialConfiguration(), configuration);
+        updater.updateConfiguration(configuration);
+        return configuration;
+    }
+
     private static void initializeConfigs(Configuration configuration)
```

Comment:
shouldn't be `configuration.set("hadoop.security.authentication", "kerberos");` this moved to hdfs conf updater now?

Reply:
why?

Reply:
This is a really tricky property. It is statically interpreted by the `UserGroupInformation`, what is really ugly. I would really love to keep everything related to static initialization here.